inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/collections/RocksDBKVBuffer.java

```diff
@@ -0,0 +1,177 @@
+        lazyGetRocksDb().put(ROCKSDB_COL_FAMILY, key, value);
+        return value;
+    }
+
+    @Override
+    public R remove(T key) {
+        checkClosed();
+        R value = get(key);
+        if (value != null) {
+            lazyGetRocksDb().delete(ROCKSDB_COL_FAMILY, (T) key);
+        }
+        return value;
+    }
+
+    @Override
+    public R get(T key) {
+        checkClosed();
+        return lazyGetRocksDb().get(ROCKSDB_COL_FAMILY, key);
+    }
+
+    /**
+     * Perform a range search according to the prefix of the key, and the KV data of all keys matching the
+     * prefix will be returned
+     * @note This stream must be closed after use, otherwise it will cause a memory leak
+     *
+     * @param keyPrefix key prefix binary data
+     * @return
+     */
+    @Override
+    public Stream<Tuple2<T, R>> scan(byte[] keyPrefix) {
+        checkClosed();
+        return lazyGetRocksDb().prefixSearch(ROCKSDB_COL_FAMILY, keyPrefix);
+    }
+
+    @Override
+    public void clear() {
+        checkClosed();
+        // todo: Here use close to mock clear.Because drop column family and recreate column family has memory leak bug.
+        // In rocksdb 5.x after dropColumnFamily and close, it will wait rocksdb release memory and disk itself.
+        // So if open a same columnFamily, it will cause memory leak. So you could not close and recreate a
+        // column family as soon as possible.In rocksdb 6.x it is ok.
+        if (null != rocksDb) {
+            LOG.info("Close rocksdb dir in {}", diskMapPath);
+            rocksDb.close();
+            cleanup(false);
+        }
+        rocksDb = null;
+    }
+
+    @Override
+    public void close() {
+        if (null != rocksDb) {
+            LOG.info("Close rocksdb dir in {}", diskMapPath);
+            rocksDb.close();
+            cleanup(false);
+        }
+        rocksDb = null;
+        closed = true;
+
+    }
+
+    private void checkClosed() {
+        Preconditions.checkArgument(!closed, "Could not operate a close RocksDB KV Buffer");
+    }
+
+    private void cleanup() {
+        this.cleanup(true);
+    }
+
+    private void addShutDownHook() {
+        this.shutdownThread = new Thread(this::cleanup);
+        Runtime.getRuntime().addShutdownHook(this.shutdownThread);
+    }
+
+    private void cleanup(boolean isTriggeredFromShutdownHook) {
+        try {
+            FileIOUtils.deleteDirectory(new File(diskMapPath));
+        } catch (IOException var3) {
+            LOG.warn("Error while deleting the disk map directory=" + this.diskMapPath, var3);
+        }
+
+        if (!isTriggeredFromShutdownHook && this.shutdownThread != null) {
+            Runtime.getRuntime().removeShutdownHook(this.shutdownThread);
+        }
+    }
+
+    private RocksDBDAO<T, R> lazyGetRocksDb() {
+        if (null == rocksDb) {
+            synchronized (this) {
+                if (null == rocksDb) {
+                    diskMapPath = String.format("%s%s%s-%s",
```

Comment:
IIUC, we should provide the settings for the user for where to store the rocksdb data ? Also some important config in case that user want to change some parameter for improvement

Reply:
User can config it by configuring the temporary directory variable when starting the java program.
In the same time , i think tmp dir is ok for buffer data storage.